crates/zed/src/languages/scala.rs

```diff
@@ -0,0 +1,61 @@
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+pub use language::*;
+use lsp::LanguageServerBinary;
+use std::{any::Any, path::PathBuf, str};
+
+pub struct MetalsLspAdapter;
+
+#[async_trait]
+impl LspAdapter for MetalsLspAdapter {
+    fn name(&self) -> LanguageServerName {
+        LanguageServerName("metals".into())
+    }
+
+    fn short_name(&self) -> &'static str {
+        "metals"
+    }
+
+    async fn fetch_latest_server_version(
+        &self,
+        _: &dyn LspAdapterDelegate,
+    ) -> Result<Box<dyn 'static + Any + Send>> {
+        Ok(Box::new(()))
+    }
+
+    async fn fetch_server_binary(
+        &self,
+        _version: Box<dyn 'static + Send + Any>,
+        _container_dir: PathBuf,
+        _: &dyn LspAdapterDelegate,
+    ) -> Result<LanguageServerBinary> {
+        Err(anyhow!(
+            "metals must be installed and available in your $PATH"
+        ))
+    }
+
+    async fn cached_server_binary(
+        &self,
+        _: PathBuf,
+        _: &dyn LspAdapterDelegate,
+    ) -> Option<LanguageServerBinary> {
+        Some(LanguageServerBinary {
+            path: "metals".into(),
+            arguments: vec![],
+        })
+    }
+
+    fn can_be_reinstalled(&self) -> bool {
+        false
+    }
+
+    async fn installation_test_binary(
```

Comment:
Maybe we could execute `metals -v` to test if the binary is properly installed (can be found on path) and runs. It should return exit code 0 and print version to the standard output (which perhaps could be used in the future to compare the installed and requested releases?). :slightly_smiling_face: